### PR TITLE
feat(plugins): add context_indicator with prompt circle + /context command

### PR DESCRIPTION
**Pull request by mmcguff**
## Summary

Adds a tiny ergonomics plugin that surfaces context-window usage:

* A colored circle in the terminal prompt right after 🐶:
  * 🟢 under 30% used
  * 🟡 30–60% used
  * 🔴 over 60% used
* A new `/context` slash command that prints a token-usage breakdown
  (messages vs. system/tools overhead vs. total / capacity) plus a
  little progress bar.

## Demo

```
🐶 🟡 puppy [code-puppy] [openai:gpt-5] (~/proj) >>>

> /context
🟡 Context usage: 50.0%
  [███████████████░░░░░░░░░░░░░░░]
  Messages : 4,500 tokens
  Overhead : 500 tokens (system prompt + tools)
  Total    : 5,000 / 10,000 tokens
  Buckets  : 🟢 <30%   🟡 30–60%   🔴 >60%
```

## Design

* **Plugin-only.** No edits to `code_puppy/command_line/` or anywhere
  in core. Uses the documented `startup`, `custom_command`, and
  `custom_command_help` hooks per `CONTRIBUTING.md`.
* **Idempotent prompt patch** on `get_prompt_with_active_model` —
  original is stashed on the module so re-installation is a no-op
  (mirrors the pattern from `prompt_newline`).
* **Token math reuses the agent's own helpers**
  (`estimate_tokens_for_message`, `_estimate_context_overhead`,
  `_get_model_context_length`) so the indicator stays in lockstep with
  the numbers the compaction subsystem already computes.
* **Defensive everywhere.** Any exception inside usage computation
  hides the indicator instead of crashing the prompt — the colored
  circle is a status badge, not a load-bearing dependency.
* **Single Responsibility split:** `usage.py` knows tokens,
  `register_callbacks.py` knows prompt-toolkit and slash commands. ~100
  lines each, well under the 600-line cap.

## Tests

`tests/plugins/test_context_indicator_plugin.py` — **22 cases** covering:

* All three threshold buckets at edge values (0%, 29.9%, 30%, 59.9%, 60%, 150%)
* `ContextUsage` proportion / percent / indicator math
* Defensive `None` returns when no agent / capacity is zero / imports fail
* Idempotent prompt patching (re-install is no-op, original is preserved)
* Indicator placement (after the 🐶 tuple, before the puppy name)
* `/context` command (success path, no-usage path, ignores unrelated names)
* Progress-bar rendering

## Files

```
code_puppy/plugins/context_indicator/
├── __init__.py              # docstring + sentinel
├── usage.py                 # token-counting + ContextUsage dataclass (~100 LOC)
└── register_callbacks.py    # prompt patch + /context command (~150 LOC)
tests/plugins/test_context_indicator_plugin.py
```

No new dependencies. No core file modifications.

### CODE REVIEW

File: tests/plugins/test_context_indicator_plugin.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    if "code_puppy.agents.agent_manager" in sys.modules:
+        return
+    stub = MagicMock()
+    stub.get_current_agent = MagicMock(side_effect=RuntimeError("unstubbed"))
+    sys.modules["code_puppy.agents.agent_manager"] = stub
+    # Also ensure parent ``code_puppy.agents`` namespace knows about it.
+    agents_pkg = sys.modules.get("code_puppy.agents")
+    if agents_pkg is not None:
+        setattr(agents_pkg, "agent_manager", stub)
+
```
[the remark this answers is not in the document]
Comment (mmcguff) [reply tests/plugins/test_context_indicator_plugin.py]: @copilot apply changes based on [this feedback](https://github.com/mpfaffenberger/code_puppy/pull/342#discussion_r3250653533)

**Comment by mpfaffenberger** [comment]
can you make sure linters / tests pass?

**Comment by mmcguff** [comment]
> can you make sure linters / tests pass?

I’ll be working on this today. I got pulled into a few other priorities earlier, but I’ll let you know as soon as it’s ready.

**Comment by mmcguff** [comment]
Addressed Copilot's review in 0c19ac3:

- **Threshold/legend mismatch** (`pick_indicator` + `/context` report): kept the threshold logic as-is (60% → 🔴 was the dev's intent per the existing test case), but updated the module comment and `/context` legend to `🟢 <30%   🟡 30–<60%   🔴 ≥60%` so the docs match the sharp `<` comparisons.
- **`get_current_usage()` partial-failure honesty**: removed the per-call try/except fallbacks. Any exception while estimating history/used/overhead/capacity now returns `None` so we hide the badge instead of surfacing a misleading 🟢. Docstring updated accordingly.
- **`sys.modules` leakage** (already handled in 417bae8 via the scoped `stub_agent_manager` fixture).
- **Tests**: added two cases covering the new strict-None paths (estimator raises / overhead raises). `ruff check .` clean, `ruff format` applied, 24/24 plugin tests pass; full suite green except 3 pre-existing failures on `main` unrelated to this PR (`test_model_factory` × 2, `browser_workflows`).
